src/rebar_dialyzer.erl

```diff
@@ -0,0 +1,259 @@
+         ]}
+       ]).
+
+plt(Config, AppFile) ->
+    PltDir = plt_dir(Config),
+    {NewConfig, RawAppName} = rebar_app_utils:app_name(Config, AppFile),
+    AppName = atom_to_list(RawAppName),
+    OtpRel = rebar_utils:otp_release(),
+    Plt = filename:join([PltDir, AppName ++ "_" ++ OtpRel ++ "_plt"]),
+    {NewConfig, Plt}.
+
+plt_dir(Config) ->
+    Location = rebar_config:get_local(Config, dialyzer_plt_location, shared),
+    plt_dir1(Config, Location).
+
+plt_dir1(_Config, Location) when is_list(Location) ->
+    case filelib:is_dir(Location) of
+        false ->
+            ?ABORT("PLT directory does not exist: ~s~n", [Location]);
+        true ->
+            Location
+    end;
+plt_dir1(_Config, shared) ->
+    {ok, Home} = init:get_argument(home),
+    filename:join([Home, ".rebar", "plt"]);
+plt_dir1(Config, local) ->
+    BaseDir = rebar_utils:base_dir(Config),
+    filename:join([BaseDir, ".rebar"]).
+
+check_plt_existence(Plt) ->
+    case filelib:is_regular(Plt) of
+        true ->
+            ok;
+        false ->
+            ?ABORT("PLT '~s' does not exist.~n"
+                   "Please run 'rebar build-plt' first.~n", [Plt])
+    end.
+
+%% dialyzer:run/1 wrapper to gracefully fail in case of Dialyzer errors
+run(Opts) ->
+    try dialyzer:run(Opts) of
+        Ws -> Ws
+    catch
+        throw:{dialyzer_error, Reason} ->
+            ?ABORT("Dialyzer error:~n~s~n", [Reason])
+    end.
+
+warnings(Config) ->
+    rebar_config:get_local(Config, dialyzer_warnings, []).
+
+filter_warnings(RawWs, Config) ->
+    RawPatterns = rebar_config:get_local(
+                    Config, dialyzer_ignored_warnings, []),
+    case RawPatterns of
+        [] ->
+            ok;
+        _ ->
+            ?CONSOLE("dialyzer_ignored_warnings is experimental and the"
+                     " config interface is subject to change~n", [])
+    end,
+    Patterns = ordsets:from_list(RawPatterns),
+    Ws = ordsets:from_list(RawWs),
+    ?DEBUG("All warnings:~n~p~n", [RawWs]),
+    ?DEBUG("Ignored warnings:~n~p~n", [RawPatterns]),
+    Intersection = ordsets:intersection(Patterns, Ws),
+    case ordsets:subtract(Patterns, Intersection) of
+        [] ->
+            ok;
+        Unused ->
+            ?WARN("Unused dialyzer_ignored_warnings entries:~n~p~n", [Unused])
+    end,
+    ordsets:subtract(Ws, Patterns).
+
+print_warnings(Ws) ->
+    lists:foreach(
+      fun(W) ->
+              ?CONSOLE("~p~n~s~n", [W, dialyzer:format_warning(W)])
+      end,
+      Ws).
+
+app_dirs(Config, AppFile) ->
+    {NewConfig, AppFileApps} = app_file_apps(Config, AppFile),
+    ?DEBUG("app file apps:~n~p~n", [AppFileApps]),
+    Deps = deps_apps(Config),
+    ?DEBUG("deps apps:~n~p~n", [Deps]),
+    ExtraApps = rebar_config:get_local(Config, dialyzer_extra_apps, []),
+    ?DEBUG("extra apps:~n~p~n", [ExtraApps]),
+    Apps = ordsets:from_list(AppFileApps ++ Deps ++ ExtraApps),
+    AppDirs = [code:lib_dir(App, ebin) || App <- ordsets:to_list(Apps)],
```

Comment:
If the lib dir is not found this will be an error tuple, `{error, bad_name}`, which results in an unhelpful error when passed to dialyzer as a directory.